lisa/schema.py

```diff
@@ -473,7 +473,18 @@ class DiskOptionSettings(FeatureSettings):
             ],
         ),
         metadata=field_metadata(
-            decoder=partial(search_space.decode_set_space_by_type, base_type=DiskType)
+            decoder=lambda input: (
+                search_space.decode_set_space_by_type(data=input, base_type=DiskType)
```

Comment:
Write a method, it would be clearer than lambda.

Reply:
updated